### PR TITLE
add rake task for remaping tags

### PR DESCRIPTION
**Pull request by andela-aadepoju**
This PR adds a rake task for remapping tags. The task will be carried out by running `rake remap_tags:execute`

### CODE REVIEW

**Comment by andela-oojewale** [comment]
👍

File: lib/tasks/remap_tags.rake [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                "queries" => "databases",
+                "deployment" => "laravel"
+                }
+
+    old_tags.each do |key, value|
+        new_tag = Tag.where(name: value)
+        ResourceTag.joins(:tag).where(tags: {name: key}).update_all(tag_id: new_tag)
+    end
+
+    tags_to_delete = ["computer", "operation", "forms", "networking", "pdo-connection",
```
Comment (andela-bebowe) [inline lib/tasks/remap_tags.rake]: I think that should be `computer science` instead of computer

File: lib/tasks/remap_tags.rake [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            "sports" =>["football"],
+            "healthcare" => ["kbl", "bupa", "health"],
+            "javascript" => ["dom", "superagent"],
+            "react" => ["react router"],
+            "ruby" => ["rspec", "serializers"],
+            "testing" => ["faker", "mock", "stub", "unit-testing"],
+            "operations" => ["leave", "leave-request", "annual-leave"],
+            "mac" => ["osx"],
+            "nairobi-fellows" => ["brown-bag", "kenya"],
+            "databases" => ["sql", "database-relation"],
```
Comment (andela-bebowe) [inline lib/tasks/remap_tags.rake]: its `database relationships` not database-relation

File: lib/tasks/remap_tags.rake [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    old_tags.each do |key, value|
+        new_tag = Tag.where(name: value)
+        ResourceTag.joins(:tag).where(tags: {name: key}).update_all(tag_id: new_tag)
+    end
+
+    tags_to_delete = ["computer", "operation", "forms", "networking", "pdo-connection",
+     "paid-courses", "programming", "associations", "facility", "ci", "mixin",
+     "discussion", "promotions", "beautiful soup"] + old_tags.keys
+
+    Tag.where(name: "orders").update_all(name: "NY-Office")
```
Comment (andela-bebowe) [inline lib/tasks/remap_tags.rake]: @andela-ooranagwa talked about using small letters all through. shouldn't this apply for NY-Office ?

File: lib/tasks/remap_tags.rake [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    tags_to_delete = ["computer science", "operation", "forms", "networking", "pdo-connection",
+     "paid-courses", "programming", "associations", "facility", "ci", "mixin",
+     "discussion", "promotions", "beautiful soup"] + old_tags.keys
+
+    Tag.where(name: "orders").update_all(name: "NY-Office")
+    Tag.where(name: "history").update_all(name: "andela-history")
+    Tag.where(name: "databases").update_all(name: "database")
+
+    Tag.where(name: tags_to_delete).destroy_all
```
Comment (andela-oakinniranye) [inline lib/tasks/remap_tags.rake]: Are you sure we really want to delete these tags? Basically, the idea of the remapping is that as we increase the number of tags, the system becomes more intelligent to do the mapping for us, which is dependent on us having a repository of existing tags being mapped to each other...

**Comment by andela-oakinniranye** [comment]
Would it be convenient to add an additional task that updates all tags with whitespaces joining them, to use dashes `-` as our conjunctions. e.g `computer science` should be `computer-science`. I think the query should just be:

```ruby
Tag.find_each do |tag|
   name = tag.name.split(' ').join('-')
   tag.update(name: name)
end
```

File: lib/tasks/remap_tags.rake [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            "sports" =>["football"],
+            "healthcare" => ["kbl", "bupa", "health"],
+            "javascript" => ["dom", "superagent"],
+            "react" => ["react router"],
+            "ruby" => ["rspec", "serializers"],
+            "testing" => ["faker", "mock", "stub", "unit-testing"],
+            "operations" => ["leave", "leave-request", "annual-leave"],
+            "mac" => ["osx"],
+            "nairobi-fellows" => ["brown-bag", "kenya"],
+            "databases" => ["sql", "database-relationships"],
```
Comment (andela-bebowe) [inline lib/tasks/remap_tags.rake]: Before this line we will have to rename `database relationship` to `database-relationships`

**Comment by andela-bebowe** [comment]
@andela-oakinniranye How often do you intend to run that line ? Thinking a validation before creating the tag might be another option

**Comment by andela-oakinniranye** [comment]
Yes, true. It would be ok for this PR to just add the callback as well. However, we still need to run this

**Comment by andela-bebowe** [comment]
So you are saying for this script you will run that and then include the validation in the tag model

**Comment by andela-oakinniranye** [comment]
I meant this PR should add the callback in the Tag model, while this script should also add that as well. Basically this script would only run once as well since we wouldn't always be having these discrepancies in the tag

**Comment by andela-bebowe** [comment]
Ok Nice

File: lib/tasks/remap_tags.rake [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    old_tags.each do |key, value|
+        new_tag = Tag.where(name: value)
+        ResourceTag.joins(:tag).where(tags: {name: key}).update_all(tag_id: new_tag)
+    end
+
+    tags_to_delete = ["computer science", "operation", "forms", "networking", "pdo-connection",
+     "paid-courses", "programming", "associations", "facility", "ci", "mixin",
+     "discussion", "promotions", "beautiful soup"] + old_tags.keys
+
+    Tag.where(name: "orders").update_all(name: "NY-Office")
```
Comment (andela-bebowe) [inline lib/tasks/remap_tags.rake]: This NY-Office ought to be ny-office ooh

Comment (andela-ooranagwa) [inline lib/tasks/remap_tags.rake]: There's a `before_save` that will take care of this.

File: lib/tasks/remap_tags.rake (hunk 4 above)
Comment (andela-aadepoju) [inline lib/tasks/remap_tags.rake]: @andela-ooranagwa what are your thoughts on this?

**Comment by andela-aadepoju** [comment]
I see @andela-ooranagwa has made the changes, thanks boss! 💯

**Pull request merged**